Sources/Views/MessageLabel.swift

```diff
@@ -23,6 +23,132 @@
  */
 
 import UIKit
+extension NSTextStorage {
+    
+    var attributedString: NSAttributedString {
+        let range = NSRange(location: 0, length: mutableString.length)
+        return attributedSubstring(from: range)
+    }
+    
+    var mutableAttributedString: NSMutableAttributedString {
+        return NSMutableAttributedString(attributedString: attributedString)
+    }
+    
+    func addAttribute(key: NSAttributedStringKey, value: AnyObject, range: NSRange? = nil) {
+        let range = range ?? NSRange(location: 0, length: length)
+        addAttribute(key, value: value, range: range)
+    }
+    
+}
+
+open class AttributeLabel: UIView {
+    
+    private lazy var layoutManager: NSLayoutManager = {
+        let manager = NSLayoutManager()
+        manager.addTextContainer(self.textContainer)
+        return manager
+    }()
+    
+    private lazy var textContainer: NSTextContainer = {
+        let container = NSTextContainer()
+        container.lineFragmentPadding = 0
```

Comment:
Should remember to use the values from the backing properties once they're added